tests/integration/targets/azure_rm_roleassignment/tasks/main.yml

```diff
@@ -1,11 +1,36 @@
 - name: setup basic facts
-  set_facts:
+  set_fact:
     uuid: bb21a88b-30e1-42b5-84e8-1d3f322de033
+    # Disk Backup Reader, unlikely to be already assigned in ansible-test resource groups.
+    az_role_definition_guid: '3e5e47e6-65f7-47ef-90b5-e5dd4d455f24'
 
 - name: List All
   azure_rm_roleassignment_info:
   register: az_role_assignments
 
+- name: Fetch specific assignments
+  azure_rm_roleassignment_info:
+    name: "{{ item.name | default(omit) }}"
+    assignee: "{{ item.assignee | default(omit) }}"
+    id: "{{ item.id | default(omit) }}"
+    role_definition_id: "{{ item.role_definition_id | default(omit) }}"
+    scope: "{{ item.scope | default(omit) }}"
+    strict_scope_match: True
+  register: az_role_assignment_specific
+  loop:
+    - name: "{{ az_role_assignments.roleassignments[0].name }}"
+      scope: "{{ az_role_assignments.roleassignments[0].scope }}"
+    - assignee: "{{ az_role_assignments.roleassignments[0].principal_id }}"
```

Comment:
> @paultaiton If get role assignments info by assignee, scope and role_definition_id, It will return role assignments more than one! It will lead line 31 fail! Thank you very much!

That was a bug that was legitimately caught by the test. I reordered the main 'module_exec' case in info module, but didn't add the assignee filter to the case that catches the assignment triplet. I just added it in latest commit, so test should pass now.